pkg/podexecutor/staticpod_windows.go

```diff
@@ -0,0 +1,345 @@
+// +build windows
+
+package podexecutor
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"github.com/Microsoft/hcsshim"
+
+	"golang.org/x/sys/windows/registry"
+
+	"github.com/rancher/k3s/pkg/cli/cmds"
+	daemonconfig "github.com/rancher/k3s/pkg/daemons/config"
+	"github.com/rancher/k3s/pkg/daemons/executor"
+	"github.com/rancher/rke2/pkg/bootstrap"
+	"github.com/rancher/rke2/pkg/images"
+	win "github.com/rancher/rke2/pkg/windows"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	"k8s.io/apiserver/pkg/authentication/authenticator"
+	"sigs.k8s.io/yaml"
+)
+
+var (
+	ssldirs = []string{
+		"/etc/ssl/certs",
+		"/etc/pki/tls/certs",
+		"/etc/ca-certificates",
+		"/usr/local/share/ca-certificates",
+		"/usr/share/ca-certificates",
+	}
+)
+
+type StaticPodConfig struct {
```

Comment:
K3s has the Embedded executor, RKE2 on Linux has StaticPod, RKE2 on Windows needs a distinct executor  (`PEBinary` perhaps) that doesn't have any copypasta carried over from the static pod executor.

Reply:
That's a good point. We should also possibly think about embedding and composition for code reuse for the similar components.